datarobot_provider/operators/feature_discovery.py

```diff
@@ -69,29 +68,24 @@ def __init__(
         self.relationships = relationships
         self.feature_discovery_settings = feature_discovery_settings
 
-    def execute(self, context: Context) -> None:
-        response = dr.client.get_client().post(
-            "/recipes/fromDataset/",
-            data={
-                "useCaseId": self.get_use_case_id(context, required=True),
-                "status": "draft",
-                "datasetId": self.dataset_id,
-                "recipeType": "FEATURE_DISCOVERY",
-            },
+    def execute(self, context: Context) -> str:
+        use_case: dr.UseCase = self.get_use_case(context, required=True)  # type: ignore[assignment]
```

Comment:
Sorry for the maybe nit.

IMHO when I see mypy ignores I feel they can likely be a code smell to possibly (in this case I think) be more defensive in the code.

It looks like this method `get_use_case()` can return `None`
So what about doing something like this instead to "keep mypy happy"?
```suggestion
        use_case = self.get_use_case(context, required=True)
        if not use_case:
            raise AirflowException("Use Case was not found please be sure you have defined a *use_case_id* parameter.")
```

Reply:
get_use_case(context, required=**True**) is doing what you've described above inside the method itself. So, we will never see `None` here even though the typehint is correct.

I can think of a bit different method `get_mandatory_use_case()` but should we?..

Reply:
Oh I see - `required` is raising an exception in `get_use_case_id()` but _not_ `get_use_case()` and I think (maybe because of walrus operator?) mypy isn't smart enough to realize that.

Hmm, I guess it's just too bad if we have to have type annotation ignore statements - I think the more we can avoid them the better 🤷

Reply:
I'll make a PR to make it more mypy-friendly. I think this PR can go as-is.